lib/puppet/provider/package/version/debian.rb

```diff
@@ -0,0 +1,171 @@
+require 'puppet/provider/package/version'
+
+class Puppet::Provider::Package::Version::Debian < Numeric
+  include Comparable
+
+  class ValidationFailure < ArgumentError; end
+
+  def self.parse(ver)
+    match, epoch, upstream_version, debian_revision = *ver.match(REGEX_FULL_RX)
+
+    raise ValidationFailure, "Unable to parse '#{ver}' as a debian version identifier" unless match
+
+    new(epoch.to_i, upstream_version, debian_revision).freeze
+  end
+
+  def self.match_digits(a)
+    a.match(/^([0-9]+)/)
+  end
+
+  def self.match_non_letters(a)
+    a.match(/^([\.\+-]+)/)
+  end
+
+  def self.match_tildes(a)
+    a.match(/^(~+)/)
+  end
+
+  def self.match_letters(a)
+    a.match(/^([A-Za-z]+)/)
+  end
+
+  def self.compare_debian_versions(mine, yours)
```

Comment:
is this going to be used outside of this class? i think this should be a private method